Lib/typing.py

```diff
@@ -433,6 +433,9 @@ def __reduce__(self):
     def __call__(self, *args, **kwds):
         raise TypeError(f"Cannot instantiate {self!r}")
 
+    def __iter__(self):
+        raise TypeError(f"{self} does not support iteration")
```

Comment:
Doesn't matter much, but the conventional language for this in a variety of similar TypeErrors in CPython (including the nearest analogue, `types.UnionType`) is "is not iterable" rather than "does not support iteration"

```suggestion
        raise TypeError(f"{self} is not iterable")
```